dev-db/cdb/files/k-r-decls-and-strict-prototypes.patch

```diff
@@ -0,0 +1,454 @@
++void 
++c (char *home, char *subdir, char *file, int uid, int gid, int mode)
+ {
+   int fdin;
+   int fdout;
+@@ -99,14 +86,8 @@ int mode;
+     strerr_die6sys(111,FATAL,"unable to chmod .../",subdir,"/",file,": ");
+ }
+ 
+-void z(home,subdir,file,len,uid,gid,mode)
+-char *home;
+-char *subdir;
+-char *file;
+-int len;
+-int uid;
+-int gid;
+-int mode;
++void 
++z (char *home, char *subdir, char *file, int len, int uid, int gid, int mode)
+ {
+   int fdout;
+ 
+@@ -137,7 +118,8 @@ int mode;
+     strerr_die6sys(111,FATAL,"unable to chmod .../",subdir,"/",file,": ");
+ }
+ 
+-main()
++int 
++main (void)
+ {
+   fdsourcedir = open_read(".");
+   if (fdsourcedir == -1)
+--- a/instcheck.c
++++ b/instcheck.c
+@@ -40,46 +40,30 @@ int mode;
+     strerr_warn6(WARNING,prefix1,prefix2,prefix3,file," has wrong type",0);
+ }
+ 
+-void h(home,uid,gid,mode)
+-char *home;
+-int uid;
+-int gid;
+-int mode;
++void 
++h (char *home, int uid, int gid, int mode)
+ {
+   perm("","","",home,S_IFDIR,uid,gid,mode);
+ }
+ 
+-void d(home,subdir,uid,gid,mode)
+-char *home;
+-char *subdir;
+-int uid;
+-int gid;
+-int mode;
++void 
++d (char *home, char *subdir, int uid, int gid, int mode)
+ {
+   if (chdir(home) == -1)
+     strerr_die4sys(111,FATAL,"unable to switch to ",home,": ");
+   perm("",home,"/",subdir,S_IFDIR,uid,gid,mode);
+ }
+ 
+-void p(home,fifo,uid,gid,mode)
+-char *home;
+-char *fifo;
+-int uid;
+-int gid;
+-int mode;
++void 
++p (char *home, char *fifo, int uid, int gid, int mode)
+ {
+   if (chdir(home) == -1)
+     strerr_die4sys(111,FATAL,"unable to switch to ",home,": ");
+   perm("",home,"/",fifo,S_IFIFO,uid,gid,mode);
+ }
+ 
+-void c(home,subdir,file,uid,gid,mode)
+-char *home;
+-char *subdir;
+-char *file;
+-int uid;
+-int gid;
+-int mode;
++void 
++c (char *home, char *subdir, char *file, int uid, int gid, int mode)
+ {
+   if (chdir(home) == -1)
+     strerr_die4sys(111,FATAL,"unable to switch to ",home,": ");
+@@ -88,20 +72,16 @@ int mode;
+   perm(".../",subdir,"/",file,S_IFREG,uid,gid,mode);
+ }
+ 
+-void z(home,file,len,uid,gid,mode)
+-char *home;
+-char *file;
+-int len;
+-int uid;
+-int gid;
+-int mode;
++void 
++z (char *home, char *file, int len, int uid, int gid, int mode)
+ {
+   if (chdir(home) == -1)
+     strerr_die4sys(111,FATAL,"unable to switch to ",home,": ");
+   perm("",home,"/",file,S_IFREG,uid,gid,mode);
+ }
+ 
+-main()
++int 
++main (void)
+ {
+   hier();
+   _exit(0);
+--- a/readwrite.h
++++ b/readwrite.h
+@@ -1,7 +1,6 @@
+ #ifndef READWRITE_H
+ #define READWRITE_H
+ 
+-extern int read();
+-extern int write();
++#include <unistd.h>
+ 
+ #endif
+--- a/trycpp.c
++++ b/trycpp.c
+@@ -1,4 +1,5 @@
+-main()
++int 
++main (void)
+ {
+ #ifdef NeXT
+   printf("nextstep\n"); exit(0);
+--- a/tryulong32.c
++++ b/tryulong32.c
+@@ -1,4 +1,5 @@
+-void main()
++void 
++main (void)
+ {
+   unsigned long u;
+   u = 1;
+--- a/x86cpuid.c
++++ b/x86cpuid.c
+@@ -1,11 +1,11 @@
+-#include <signal.h>
+-
+-void nope()
++__END_DECLS void 
```

Comment:
this _END_DECLS thing looks suspicious, what's that about?

Reply:
cproto kindly added it and I didn't realize, dropping